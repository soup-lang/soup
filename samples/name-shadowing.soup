func nested(a, b, cb)
    setTimeout(%, 0)
    cb(a + b, a * b)

a: [0, 1, 2, 3]
b: a[1, 3]

a.push(-1)
b.push(-2)

x: a[4]
y: b[2]

if a[0] < b[0]
    nested(a[1], b[1], %(x, y))
    console.log(x, y)

console.log(x, y)

try
    x: 10
    y: 20
    console.log(x, y)
    throw x + y
catch e
    x: e * e
    y: x - e
    console.log(x, y)

console.log(x, y)
