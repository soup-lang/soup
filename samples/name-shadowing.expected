3 2
-1 -2
10 20
900 870
-1 -2
